pixeltable/catalog/catalog.py

```diff
@@ -171,9 +184,19 @@ def begin_xact(self, *, tbl_id: Optional[UUID] = None, for_write: bool = False)
         or metadata.
         """
         if Env.get().in_xact:
+            if tbl_id is not None and for_write:
+                # make sure that we requested the required table lock at the beginning of the transaction
+                assert tbl_id == self._x_locked_tbl_id, f'{tbl_id} != {self._x_locked_tbl_id}'
```

Comment:
(this isn't a blocking comment for the PR but something to think about for the future) I wonder if there are scenarios where we might need xlocks for multiple tables in the same transaction? If possible, it'd be more flexible to acquire additional xlocks as needed, rather than assert as here

Reply:
Added explanation why that doesn't work.